modules/csg/csg_shape.cpp

```diff
@@ -483,6 +483,46 @@ void CSGShape3D::_update_collision_faces() {
 	}
 }
 
+Ref<ArrayMesh> CSGShape3D::bake_static_mesh() {
+	Ref<ArrayMesh> baked_mesh;
+	if (is_root_shape() && root_mesh.is_valid()) {
+		baked_mesh = root_mesh;
+	} else {
+		baked_mesh.instantiate();
+	}
+	return baked_mesh;
+}
+
+Ref<ConcavePolygonShape3D> CSGShape3D::bake_collision_shape() {
+	Ref<ConcavePolygonShape3D> baked_collision_shape;
+	if (is_root_shape() && root_collision_shape.is_valid()) {
+		baked_collision_shape.instantiate();
+		baked_collision_shape->set_faces(root_collision_shape->get_faces());
+	} else if (is_root_shape()) {
+		CSGBrush *n = _get_brush();
+		ERR_FAIL_NULL_V_MSG(n, baked_collision_shape, "Cannot get CSGBrush.");
+		Vector<Vector3> physics_faces;
+		physics_faces.resize(n->faces.size() * 3);
+		Vector3 *physicsw = physics_faces.ptrw();
+
+		for (int i = 0; i < n->faces.size(); i++) {
+			int order[3] = { 0, 1, 2 };
+
+			if (n->faces[i].invert) {
+				SWAP(order[1], order[2]);
+			}
+
+			physicsw[i * 3 + 0] = n->faces[i].vertices[order[0]];
+			physicsw[i * 3 + 1] = n->faces[i].vertices[order[1]];
+			physicsw[i * 3 + 2] = n->faces[i].vertices[order[2]];
+		}
+
+		baked_collision_shape.instantiate();
+		baked_collision_shape->set_faces(physics_faces);
```

Comment:
This duplicates code from `_update_collision_faces()`. It should be unified.

Reply:
moved to new `_get_brush_collision_faces()` function.